scripts/infill_costs.py

```diff
@@ -0,0 +1,1120 @@
+# -*- coding: utf-8 -*-
+"""Script to infill cost matrices produced by OTP4GB-py."""
+
+##### IMPORTS #####
+from __future__ import annotations
+
+import argparse
+import datetime
+import enum
+import functools
+import pathlib
+import re
+import sys
+import textwrap
+from typing import Any, Callable, Mapping, NamedTuple, Optional, Union
+
+import caf.toolkit
+import numpy as np
+import pandas as pd
+import pydantic
+import tqdm
+from matplotlib import figure, offsetbox, patches
+from matplotlib import pyplot as plt
+from matplotlib.backends import backend_pdf
+from pydantic import dataclasses, types
+from scipy import optimize
+
+sys.path.extend((".", ".."))
+import otp4gb
+from otp4gb import centroids, config, cost, logging, parameters
+
+##### CONSTANTS #####
+LOG = logging.get_logger(otp4gb.__package__ + ".infill_costs")
+MARKER_SIZE = 10
+
+
+##### CLASSES #####
+@dataclasses.dataclass
+class InclusiveRange:
+    """Inclusive range iterator."""
+
+    start: int
+    end: int
+
+    _current: int = 0
+
+    def __iter__(self):
+        self._current = self.start - 1
+        return self
+
+    def __next__(self) -> int:
+        self._current += 1
+        if self._current <= self.end:
+            return self._current
+
+        raise StopIteration()
+
+
+class InfillMethod(enum.StrEnum):
+    """Cost infilling methods."""
+
+    MEAN_RATIO = enum.auto()
+    LINEAR = enum.auto()
+    POLYNOMIAL_2 = enum.auto()
+    POLYNOMIAL_3 = enum.auto()
+    POLYNOMIAL_4 = enum.auto()
+    EXPONENTIAL = enum.auto()
+    LOGARITHMIC = enum.auto()
+    X_PLUS_LOGX = enum.auto()
+
+
+class OutlierMethod(enum.StrEnum):
+    """Method for calculating outliers in `CurveFit`."""
+
+    STDDEV = enum.auto()
+    GEH = enum.auto()
+
+
+class InfillParameters(caf.toolkit.BaseConfig):
+    """Config for `infill_costs` module."""
+
+    folders: list[types.DirectoryPath]
+    infill_columns: dict[str, float]
+    zero_cost_zones: InclusiveRange
+    infill_methods: list[InfillMethod]
+    outlier_method: Optional[OutlierMethod] = None
+    outlier_cutoff: Optional[float] = None
+
+
+@dataclasses.dataclass
+class InfillArgs:
```

Comment:
What is the difference between this and CombineCostArgs combine_costs.py? I assume this can't be borrowed from combine_costs.py with an import